backend/sirius-web-spring-collaborative/src/main/java/org/eclipse/sirius/web/spring/collaborative/projects/EditingContextEventProcessorRegistry.java

```diff
@@ -86,35 +94,54 @@ public EditingContextEventProcessorRegistry(IEditingContextSearchService editing
     public Optional<IEditingContextEventProcessor> getOrCreateEditingContextEventProcessor(UUID editingContextId) {
         Optional<IEditingContextEventProcessor> optionalEditingContextEventProcessor = Optional.empty();
         if (this.editingContextSearchService.existsById(editingContextId)) {
-            EditingContextEventProcessor editingContextEventProcessor = this.editingContextEventProcessors.get(editingContextId);
-            if (editingContextEventProcessor == null) {
+            optionalEditingContextEventProcessor = Optional.ofNullable(this.editingContextEventProcessors.get(editingContextId))
+                    .map(EditingContextEventProcessorEntry::getEditingContextEventProcessor);
+            if (optionalEditingContextEventProcessor.isEmpty()) {
                 Optional<IEditingContext> optionalEditingContext = this.editingContextSearchService.findById(editingContextId);
                 if (optionalEditingContext.isPresent()) {
                     IEditingContext editingContext = optionalEditingContext.get();
-                    editingContextEventProcessor = new EditingContextEventProcessor(editingContext, this.editingContextPersistenceService, this.applicationEventPublisher, this.objectService,
+
+                    var editingContextEventProcessor = new EditingContextEventProcessor(editingContext, this.editingContextPersistenceService, this.applicationEventPublisher, this.objectService,
                             this.editingContextEventHandlers, this.representationEventProcessorComposedFactory);
-                    this.editingContextEventProcessors.put(editingContextId, editingContextEventProcessor);
+                    Disposable subscription = editingContextEventProcessor.canBeDisposed().delayElements(Duration.ofSeconds(30)).subscribe(canBeDisposed -> {
```

Comment:
Magic number: should be at least a constant, maybe even configurable via a system property (I believe Spring supports properties which denote times with units).